Upgrade to 0.6.0

Lestropie opened the pull request:
Had to address #123 due to it precluding processing of data in my own lab.

Results in increment to version `0.6.0` as the incorporation of `dwicat` will result in a change in behaviour even for data that previously would have processed to completion.

On *MRtrix3* I have created a branch off of `3.0.4` with a single commit that cherry-picks https://github.com/MRtrix3/mrtrix3/pull/2702 off of `dev`. Makes the behaviour of that software dependency a little more reportable.

@aaroncapon Do you want to try building a container with this code (and the updated *MRtrix3* dependency) and make sure it runs before I merge & tag? CI is completely broken here, repo is highly neglected, so independent verification would be good if not onerous.

Lestropie committed:
Downgrade complex data rescale warning

Lestropie committed:
Make DWI import robust to additional dots in filesystem path

Lestropie committed:
Incorporate dwicat

- Make use of MRtrix3 dwicat command, which means that multiple input DWI series will have their intensities balanced based on their relative b=0 intensities.
In addition, if the input DWIs have different voxel grids, they will be conformed onto the same voxel grid prior to running dwifslpreproc; this will be done using rigid-body transformation if possible.
- In addition, if contents of the fmap/ directory are to be used, they will be explicitly aligned to the DWI voxel grid for construction of the SE-EPI image to be used as input to dwifslpreproc, avoiding interpolation if possible.

Lestropie committed:
Version increment to 0.6.0

aaroncapon commented:
Hi. Thanks for all this work @Lestropie!
I should get a chance to build the container today or tomorrow. Will report back with the results.

aaroncapon commented:
Unfortunately, there was an issue during processing using the new code. Below is the error message.

> Command:  mrconvert dwi2_denoised_mag_degibbs.mif dwi2_denoised_mag_degibbs_bzero.mif -coord 3 0,1
> Command:  mrcat [dwi*_denoised_mag_degibbs_bzero.mif (2 items)] dwifslpreproc_seepi.mif -axis 3
> mrtrix3_connectome.py: Performing various geometric corrections of DWIs
> Command:  dwifslpreproc dwifslpreproc_in.mif dwifslpreproc_out.mif -se_epi dwifslpreproc_seepi.mif -eddy_options " --repol" -rpe_header -eddyqc_text eddyqc/
> Command:  mrmath dwifslpreproc_out.mif max -axis 3 - | mrthreshold - dwi_validdata_mask.mif -abs 0.0 -comparison gt
> mrtrix3_connectome.py: Performing intial DWI brain masking
> Command:  dwi2mask dwifslpreproc_out.mif dwi_mask_init.mif
> 
> mrtrix3_connectome.py: [ERROR] dwi2mask dwifslpreproc_out.mif dwi_mask_init.mif (mrtrix3_connectome.py:1508)
> mrtrix3_connectome.py: [ERROR] Information from failed command:
> mrtrix3_connectome.py:
>                    
> Error: argument algorithm: invalid choice: 'dwifslpreproc_out.mif' (choose from '3dautomask', 'ants', 'b02template', 'consensus', 'fslbet', 'hdbet', 'legacy', 'mean', 'mtnorm', 'synthstrip', 'trace')
>   Usage: dwi2mask algorithm [ options ] ...
> (Run dwi2mask -help for more information)
>                      
> mrtrix3_connectome.py:
> mrtrix3_connectome.py: [ERROR] For debugging, inspect contents of scratch directory: /dwi_scratch/dwi2adc/9UG34Y/mrtrix3_connectome.py-tmp-GFCYVU/
> mrtrix3_connectome.py: Scratch directory retained; location: /dwi_scratch/dwi2adc/9UG34Y/mrtrix3_connectome.py-tmp-GFCYVU


As I said yesterday, this can be low priority as the subject that encountered this issue has already been sent for assessment.

Lestropie committed:
dwifslpreproc: Use --data_is_shelled on unknown failure

An instance was discovered where FSL eddy fails due to judging the DWI data to not be a shelled acquisition, but the contents that it writes to stdout is some form of binary data rather than the expected error meswsage.
With this change, if eddy fails due to an unknwon reason, but it has not yet been attempted with this option included, that second attempt will be made, even though the corresponding error message has not yet been encountered.

Lestropie committed:
Fix exception catch on mrinfo shell detection failure

Lestropie committed:
Change MRtrix3 version for dwicat update

In d4b25b38095c3177e8e65f865721a056c03bd6a5, the version of MRtrix3 was updated to introduce a feature addition to dwicat. However, this involved changing that dependency to the development branch of MRtrix3, which contains changes over and above just the enhancement of dwicat.
With this commit, the version of MRtrix3 installed is genuinely 3.0.4, with some additional unrelated hotfixes since that tag, plus the cherry-picked enhancement to dwicat, such that no other major changes to MRtrix3 are introduced.

Lestropie committed:
Update MRtrix3 version

New version is 3.0.5 but with two additional changes:
- Hotfix for -export_grad_fsl (https://github.com/MRtrix3/mrtrix3/pull/3112).
- Enhancements to dwicat back-ported from development brain to 3.0.x (https://github.com/MRtrix3/mrtrix3/pull/2702).

Lestropie committed:
Fixes to container recipes for 0.6.0
